lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+              :registered_vehicles,
+              :collected_fees,
+              :license_data
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_info)
+    @name = facility_info[:name]
+    @address = facility_info[:address]
+    @phone = facility_info[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+  
+  def register_vehicle(vehicle)
+    if @services.include?("Vehicle Registration")
+      @registered_vehicles << vehicle
+      vehicle.registration_date = Date.today
+      
+      if vehicle.antique?
+        @collected_fees += 25
+        vehicle.plate_type = :antique
+      elsif vehicle.electric_vehicle?
+        @collected_fees += 200
+        vehicle.plate_type = :ev
+      else
+        @collected_fees += 100
+        vehicle.plate_type = :regular
+      end
+    end
```

Comment:
You could use an explicit return for a guard clause here, which would eliminate the need for a nested conditional:
```ruby
return false unless @services.include?("Vehicle Registration")
@registered_vehicles << vehicle
# etc
```